dlpy/applications/vgg.py

```diff
@@ -296,13 +298,19 @@ def VGG16(conn, model_table='VGG16', n_classes=1000, n_channels=3, width=224, he
     # get all the parms passed in
     parameters = locals()
 
+    # check the type
+    check_layer_class(reshape_after_input, Reshape)
+
     if not pre_trained_weights:
         model = Sequential(conn=conn, model_table=model_table)
 
         # get the input parameters
         input_parameters = get_layer_options(input_layer_options, parameters)
         model.add(InputLayer(**input_parameters))
 
+        if reshape_after_input:
+            model.add(reshape_after_input)
```

Comment:
there is already instance of function - you don't need another api for this. please add a check right here so that you don't need to check if reshape_after_input is not none. thanks

Reply:
Pushed the changes to use type().